kiwi/iso_tools/xorriso.py

```diff
@@ -85,23 +89,33 @@ def init_iso_creation_parameters(self, custom_args=None):
         self.iso_parameters += [
             '-joliet', 'on', '-padding', '0'
         ]
-        loader_file = self.boot_path + '/loader/isolinux.bin'
 
-        syslinux_lookup_paths = [
-            '/usr/share/syslinux', '/usr/lib/syslinux/modules/bios',
-            '/usr/lib/ISOLINUX'
-        ]
-        mbr_file = Path.which('isohdpfx.bin', syslinux_lookup_paths)
-        if not mbr_file:
-            raise KiwiIsoToolError(
-                'isohdpfx.bin not found in {0}'.format(syslinux_lookup_paths)
-            )
+        if Defaults.is_x86_arch(self.arch):
+            if efi_mode:
+                loader_file = self.boot_path + '/loader/eltorito.img'
+                mbr_file = os.sep.join(
+                    [self.source_dir, self.boot_path, '/loader/boot_hybrid.img']
+                )
+                self.iso_loaders += [
+                    '-boot_image', 'grub', 'bin_path={0}'.format(loader_file),
+                    '-boot_image', 'grub', 'grub2_mbr={0}'.format(mbr_file),
+                    '-boot_image', 'grub', 'grub2_boot_info=on'
+                ]
+            else:
+                loader_file = self.boot_path + '/loader/isolinux.bin'
+                syslinux_lookup_paths = [
```

Comment:
shouldn't we use here the list `syslinux_dirs` that is also defined in `IsoToolsBase`. Probably we should make that list an attribute of the base class, this way we can easily reuse it.

Reply:
yes good point. part of the refactoring was to move the setup of the media dir to the base class as it's independent of the used tooling. That provides the lookup paths and I agree we could now also nicely use in the xorriso implementation.

I'll update the code.

However I'm currently discussing an obs issue with Adrian and Michael regarding Fedora builds and there seems to be something very very weird on the obs side. I'd like to get that sorted out before I update the code.

Reply:
@davidcassany I pushed a commit to address your issue. Should be good now